core/src/main/java/hudson/model/AbstractBuild.java

```diff
@@ -742,13 +742,19 @@ protected final boolean performAllBuildSteps(BuildListener listener, Iterable<?
         }
 
         private void reportError(BuildStep bs, Throwable e, BuildListener listener, boolean phase) {
-            final String publisher = ((Publisher) bs).getDescriptor().getDisplayName();
+            final String buildStep;
+
+            if (bs instanceof Publisher) {
+                buildStep = ((Publisher) bs).getDescriptor().getDisplayName();
+            } else {
+                buildStep = bs.getClass().getName();
+            }
 
             if (e instanceof AbortException) {
-                LOGGER.log(Level.FINE, "{0} : {1} failed", new Object[] {AbstractBuild.this, publisher});
-                listener.error("Publisher '" + publisher + "' failed: " + e.getMessage());
+                LOGGER.log(Level.FINE, "{0} : {1} failed", new Object[] {AbstractBuild.this, buildStep});
+                listener.error(buildStep + "' failed: " + e.getMessage());
```

Comment:
:ant: this will produce a weirdly formatted message:

    Something' failed: file not found